velox/exec/GroupingSet.cpp

```diff
@@ -714,6 +714,85 @@ void GroupingSet::updateRow(SpillMergeStream& input, char* FOLLY_NONNULL row) {
   }
   mergeSelection_.setValid(input.currentIndex(), false);
 }
+void GroupingSet::abandonPartialAggregation() {
+  abandonedPartialAggregation_ = true;
+  allHaveToIntermediate_ = true;
+  for (auto& aggregate : aggregates_) {
+    if (!aggregate->supportsToIntermediate()) {
+      allHaveToIntermediate_ = false;
+    }
+  }
+  if (!allHaveToIntermediate_) {
+    intermediateRows_ = table_->moveRows();
+    intermediateRows_->clear();
+  }
+  table_ = nullptr;
+}
+
+void GroupingSet::toIntermediate(
+    const RowVectorPtr& input,
+    RowVectorPtr& result) {
+  VELOX_CHECK(abandonedPartialAggregation_);
+  VELOX_CHECK(result.unique());
+  if (!isRawInput_) {
```

Comment:
I think we should VELOX_CHECK(isRawInput) instead of returning because returning will mask the problem?

Secondarily, what about Step::kIntermediate aggregation? isRawInput_ would be false, but maybe still it can benefit from disable hashing.